src/Common/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -10,7 +10,7 @@ internal static partial class TypeUtils
     {
         public static bool AreEquivalent(Type t1, Type t2)
         {
-            return t1.IsEquivalentTo(t2);
+            return t1 != null && t1.IsEquivalentTo(t2);
```

Comment:
Type implements "==" operator. We just want a null check.
Would it be better to do something like:   `return t1?.IsEquivalentTo(t2) ?? false`